napari/_app_model/context/_layerlist_context.py

```diff
@@ -122,7 +122,7 @@ def _active_shape(s: LayerSel) -> Optional[Tuple[int, ...]]:
 
 
 def _same_shape(s: LayerSel) -> bool:
-    return len({getattr(x.data, "shape", ()) for x in s}) == 1
+    return len({tuple(getattr(x.data, "shape", ())) for x in s}) == 1
```

Comment:
Please add a comment with an explanation why tuple is required. What type has `mlx` or `jax` `shape`

Reply:
Good point! I've added a docstring with a corresponding note.

Reply:
which libraries return list?

Reply:
Only mlx *that I know of*. I also just saw that this is actually against the array-API spec, which [specifies that the .shape attribute must be immutable](https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.shape.html). However, this is a low-cost modification to add compatibility with current library versions, so I suggest leaving this in. I'll add a comment that we can remove the tuple cast once ~all "major" array libraries return a tuple.

Reply:
done

Reply:
Wonder why they did that--maybe worth opening an issue?